nms/app/components/context/RefreshContext.ts

```diff
@@ -9,55 +9,80 @@
 import {
   FetchFegGateways,
   getActiveFegGatewayId,
   getFegGatewaysHealthStatus,
-  // $FlowFixMe migrated to typescript
 } from '../../state/feg/EquipmentState';
-// $FlowFixMe[cannot-resolve-module] for TypeScript migration
-// $FlowFixMe[cannot-resolve-module] for TypeScript migration
 import {FetchFegSubscriberState} from '../../state/feg/SubscriberState';
-// $FlowFixMe[cannot-resolve-module] for TypeScript migration
 import {FetchSubscriberState} from '../../state/lte/SubscriberState';
+import {GatewayId} from '../../../shared/types/network';
 import {useContext, useEffect, useRef, useState} from 'react';
-import type {EnqueueSnackbarOptions} from 'notistack';
+import type {OptionsObject} from 'notistack';
 
 export const REFRESH_INTERVAL = 30000;
 
-export const RefreshTypeEnum = Object.freeze({
+export const RefreshTypeEnum = {
   SUBSCRIBER: 'subscriber',
   GATEWAY: 'gateway',
   FEG_GATEWAY: 'feg_gateway',
   FEG_SUBSCRIBER: 'feg_subscriber',
   ENODEB: 'enodeb',
-});
-
-type Props = {
-  context: typeof React.Context,
-  networkId: string,
-  type: RefreshType,
-  interval?: number,
-  id?: string,
+} as const;
+
+type ContextMap = {
+  [RefreshTypeEnum.SUBSCRIBER]: typeof SubscriberContext;
+  [RefreshTypeEnum.GATEWAY]: typeof GatewayContext;
+  [RefreshTypeEnum.FEG_GATEWAY]: typeof FEGGatewayContext;
+  [RefreshTypeEnum.FEG_SUBSCRIBER]: typeof FEGSubscriberContext;
+  [RefreshTypeEnum.ENODEB]: typeof EnodebContext;
+};
+
+type StateMap = {
+  [RefreshTypeEnum.SUBSCRIBER]: SubscriberContextType['sessionState'];
+  [RefreshTypeEnum.GATEWAY]: GatewayContextType['state'];
+  [RefreshTypeEnum.FEG_GATEWAY]: {
+    fegGateways: FEGGatewayContextType['state'];
+    health: FEGGatewayContextType['health'];
+    activeFegGatewayId: FEGGatewayContextType['activeFegGatewayId'];
+  };
+  [RefreshTypeEnum.FEG_SUBSCRIBER]: FEGSubscriberContextType['sessionState'];
+  [RefreshTypeEnum.ENODEB]: EnodebContextType['state'];
+};
+
+type RefreshType = keyof ContextMap;
+
+type Props<T extends RefreshType> = {
+  type: T;
+  context: ContextMap[T];
+  networkId: string;
+  interval?: number;
+  id?: string;
   enqueueSnackbar?: (
     msg: string,
-    cfg: EnqueueSnackbarOptions,
-  ) => ?(string | number),
-  refresh: boolean,
-  lastRefreshTime?: string,
+    cfg: OptionsObject,
+  ) => string | number | undefined | null;
+  refresh: boolean;
+  lastRefreshTime?: string;
 };
 
-type RefreshType = $Values<typeof RefreshTypeEnum>;
-
-export function useRefreshingContext(props: Props) {
-  const ctx = useContext(props.context);
+// TODO: This hook is not well designed and typing it correctly is nearly impossible,
+//  it should be replaced with a simpler solution.
+/* eslint-disable @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-return */
```

Comment:
Would this be a `TODO[ts-migration]`?

Reply:
No I intentionally omitted the `[ts-migration]` because this is a design issue not a TypeScript migration issue.